src/priceHistory/priceHistoryChart.tsx

```diff
@@ -0,0 +1,357 @@
+    if (!highPrice || priceHistoryPrices[i].priceUsd > highPrice.priceUsd) {
+      highPrice = priceHistoryPrices[i]
+    }
+    if (!lowPrice || priceHistoryPrices[i].priceUsd < lowPrice.priceUsd) {
+      lowPrice = priceHistoryPrices[i]
+    }
+
+    // Only grab one price per chart step & the most recent price
+    if (
+      lastTimestampAdded &&
+      priceHistoryPrices[i].priceFetchedAt - lastTimestampAdded <
+        ONE_HOUR_IN_MILLIS * CHART_STEP_IN_HOURS &&
+      i !== priceHistoryPrices.length - 1
+    ) {
+      continue
+    } else {
+      lastTimestampAdded = priceHistoryPrices[i].priceFetchedAt
+      const { priceUsd } = priceHistoryPrices[i]
+      const localAmount = dollarsToLocal(priceUsd)
+      chartData.push({
+        amount: localAmount ? localAmount.toNumber() : 0,
+        displayValue: localAmount ? displayLocalCurrency(localAmount) : '',
+        priceFetchedAt: priceHistoryPrices[i].priceFetchedAt,
+      })
+    }
+  }
+
+  // Make sure the highest and lowest prices are included in the chart
+  if (highPrice) {
+    chartData.push({
+      amount: dollarsToLocal(highPrice.priceUsd)?.toNumber() || 0,
+      displayValue: displayLocalCurrency(dollarsToLocal(highPrice.priceUsd) || 0),
+      priceFetchedAt: highPrice.priceFetchedAt,
+    })
+  }
+  if (lowPrice) {
+    chartData.push({
+      amount: dollarsToLocal(lowPrice.priceUsd)?.toNumber() || 0,
+      displayValue: displayLocalCurrency(dollarsToLocal(lowPrice.priceUsd) || 0),
+      priceFetchedAt: lowPrice.priceFetchedAt,
+    })
+  }
+
+  // Sort the chart data by timestamp and remove duplicates
+  const sortedChartData = chartData.sort((a, b) => a.priceFetchedAt - b.priceFetchedAt)
+  const uniqueChartData = uniqBy(sortedChartData, 'priceFetchedAt')
+
+  const RenderPoint = renderPointOnChart(uniqueChartData, CHART_WIDTH, color)
+
+  const values = uniqueChartData.map((el) => el.amount)
+  const min = Math.min(...values)
+  const max = Math.max(...values)
+  let domain
+  if (max - min < CHART_MIN_VERTICAL_RANGE) {
+    // Use min of the CHART_MIN_VERTICAL_RANGE or 1% of the min value
+    // This works better for small values e.g. ImpactMarket
+    const offset = Math.min(CHART_MIN_VERTICAL_RANGE - (max - min) / 2, min / 100)
+    domain = {
+      y: [min - offset, max + offset] as [number, number],
+      x: [0, uniqueChartData.length - 1] as [number, number],
+    }
+  }
+
+  const latestTimestamp = uniqueChartData.at(-1)?.priceFetchedAt
+  const earliestTimestamp = uniqueChartData.at(0)?.priceFetchedAt
+
+  return (
+    <View style={[styles.container, containerStyle]} testID={testID}>
+      <VictoryGroup
+        domainPadding={CHART_DOMAIN_PADDING}
+        singleQuadrantDomainPadding={false}
+        padding={{ left: chartPadding, right: chartPadding }}
+        width={CHART_WIDTH}
+        height={CHART_HEIGHT}
+        data={uniqueChartData.map((el) => el.amount)}
+        domain={domain}
+      >
+        {/* @ts-ignore */}
```

Comment:
could the `@ts-ignore`'s be avoided? If not, could we document why this is required and maybe use `@ts-expect-error` instead?

Reply:
It looks like it's needed as we dynamically create the datum. Switched to `@ts-expect-error` and added a note in 5f2ba7b5069f4e4da2fb07bf176179e24f1fe956.